mbed_targets/_internal/targets_json_parsers/overriding_attribute_parser.py

```diff
@@ -93,11 +113,21 @@ def _determine_overridden_attributes(override_order: List[dict]) -> Dict[str, An
     Returns:
         A dictionary containing all the overridden attributes for a target
     """
-    target_attributes = reduce(lambda x, y: {**x, **y}, reversed(override_order))
+    target_attributes = _reduce_right_list_of_dictionaries(override_order)
+    for merging_attribute in MERGING_ATTRIBUTES:
+        override_order_for_single_attribute = map(lambda x: cast(dict, x.get(merging_attribute, {})), override_order)
+        merged_attribute_elements = _reduce_right_list_of_dictionaries(list(override_order_for_single_attribute))
```

Comment:
You might lose additional config option attributes here. I'm not sure what level of overrides this operates on, but isn't it possible that:
```py
source_1 = {
  "config": {
    "variable_1": {
      "help": "help text",
      "value": 123
    }
  }
}
config_2 = {
 "config": {
   "variable_1": 456
  }
}
```
Merging 2 into 1 will blat "help"? Do I understand this correctly?

Reply:
Yeah it will. Thats a good point, I had assumed that the merging was only needed at that one level but maybe its all the way down 🤔 I will check in the existing tools.

Reply:
Yes, this is ok. There shouldn't be any merging of config params, they just overwrite.  The example you give above is actually handled with the 'overrides' field which I will make a separate PR to handle.